impeller/entity/contents/vertices_contents.cc

```diff
@@ -270,15 +84,30 @@ void VerticesSimpleBlendContents::SetEffectTransform(Matrix transform) {
   inverse_matrix_ = transform.Invert();
 }
 
+void VerticesSimpleBlendContents::SetLazyTexture(
+    const LazyTexture& lazy_texture) {
+  lazy_texture_ = lazy_texture;
+}
+
 bool VerticesSimpleBlendContents::Render(const ContentContext& renderer,
                                          const Entity& entity,
                                          RenderPass& pass) const {
-  FML_DCHECK(texture_);
+  FML_DCHECK(texture_ || lazy_texture_ ||
+             blend_mode_ == BlendMode::kDestination);
   BlendMode blend_mode = blend_mode_;
   if (!geometry_->HasVertexColors()) {
     blend_mode = BlendMode::kSource;
   }
 
+  std::shared_ptr<Texture> texture;
+  if (blend_mode != BlendMode::kDestination) {
+    if (!texture_) {
+      texture = lazy_texture_(renderer);
+    } else {
+      texture = texture_;
+    }
+  }
```

Comment:
Nit: If the texture is nullptr here, set it to the empty texture right away so you don't have to do all the null checks later? Stuff like the size, y scale coord, etc. will just work without additional checks.

Reply:
Ahh good idea 👍

Reply:
Done